File: Samples/visio_managed_code_interop_vs2010/Visio2010_VB_Samples/Shape_SetResults.vb
Imports IVisio = Microsoft.Office.Interop.Visio


Partial Public Class VS2010_VB_Samples

    Shared Sub Shape_SetResults(ByVal doc As Microsoft.Office.Interop.Visio.Document)
        Dim page = VisioInterop.Util.CreateStandardPage(doc, "SSR")
        Dim shape = VisioInterop.Util.CreateStandardShape(page)

        ' CREATE REQUEST
        Dim request = {New With { _
                Key .Section = CShort(IVisio.VisSectionIndices.visSectionObject), _
                Key .Row = CShort(IVisio.VisRowIndices.visRowXFormOut), _
                Key .Cell = CShort(IVisio.VisCellIndices.visXFormWidth), _
                Key .UnitCode = CShort(IVisio.VisUnitCodes.visNoCast), _
                Key .Result = CDbl(8.2) _
                }, _
                       New With { _
                Key .Section = CShort(IVisio.VisSectionIndices.visSectionObject), _
                Key .Row = CShort(IVisio.VisRowIndices.visRowXFormOut), _
                Key .Cell = CShort(IVisio.VisCellIndices.visXFormHeight), _
                Key .UnitCode = CShort(IVisio.VisUnitCodes.visNoCast), _
                Key .Result = CDbl(1.3) _
                }}

        ' MAP THE REQUEST TO THE STRUCTURES VISIO EXPECTS
        Dim SRCStream = New Short(request.Length * 3 - 1) {}
        Dim results_objects = New Object(request.Length - 1) {}
        Dim unitcodes = New Object(request.Length - 1) {}
        For i As Integer = 0 To request.Length - 1
            SRCStream((i * 3) + 0) = request(i).Section
            SRCStream((i * 3) + 1) = request(i).Row
            SRCStream((i * 3) + 2) = request(i).Cell
            results_objects(i) = request(i).Result
            unitcodes(i) = request(i).UnitCode
        Next

        ' EXECUTE THE REQUEST
        Dim flags As Short = 0
        Dim count As Integer = shape.SetResults(SRCStream, unitcodes, results_objects, flags)

        ' DISPLAY THE INFORMATION
        shape.Text = "SetResults"

    End Sub

End Class

File: VisioCSharpSamples/VisioCSharpSamples/shape_setresults.cs
using IVisio = Microsoft.Office.Interop.Visio;


namespace VisioCSharpSamples
{

    public static partial class Samples
    {
        public static void Shape_SetResults(IVisio.Document doc)
        {
            var page = Util.CreateStandardPage(doc, "SSR");
            var shape = Util.CreateStandardShape(page);

            // CREATE REQUEST
            var request = new[]
                              {
                                  new
                                      {
                                          Section = (short) IVisio.VisSectionIndices.visSectionObject,
                                          Row = (short) IVisio.VisRowIndices.visRowXFormOut,
                                          Cell = (short) IVisio.VisCellIndices.visXFormWidth,
                                          UnitCode = (short) IVisio.VisUnitCodes.visNoCast,
                                          Result = (double) 8.2
                                      },
                                  new
                                      {
                                          Section = (short) IVisio.VisSectionIndices.visSectionObject,
                                          Row = (short) IVisio.VisRowIndices.visRowXFormOut,
                                          Cell = (short) IVisio.VisCellIndices.visXFormHeight,
                                          UnitCode = (short) IVisio.VisUnitCodes.visNoCast,
                                          Result = (double) 1.3
                                      }
                              };

            // MAP THE REQUEST TO THE STRUCTURES VISIO EXPECTS
            var SRCStream = new short[request.Length*3];
            var results_objects = new object[request.Length];
            var unitcodes = new object[request.Length];
            for (int i = 0; i < request.Length; i++)
            {
                SRCStream[(i*3) + 0] = request[i].Section;
                SRCStream[(i*3) + 1] = request[i].Row;
                SRCStream[(i*3) + 2] = request[i].Cell;
                results_objects[i] = request[i].Result;
                unitcodes[i] = request[i].UnitCode;
            }

            // EXECUTE THE REQUEST
            short flags = 0;
            int count = shape.SetResults(SRCStream, unitcodes, results_objects, flags);

            // DISPLAY THE INFORMATION
            shape.Text = "SetResults";
        }
    }
}